core/action/service.go

```diff
@@ -46,7 +46,8 @@ func (s Service) Get(ctx context.Context, id string) (Action, error) {
 func (s Service) Create(ctx context.Context, action Action) (Action, error) {
 	currentUser, err := s.userService.FetchCurrentUser(ctx)
 	if err != nil {
-		return Action{}, fmt.Errorf("%w: %s", user.ErrInvalidEmail, err.Error())
+		email, _ := user.GetEmailFromContext(ctx)
```

Comment:
Rather than doing this, we could enrich error detail inside `FetchCurrentUser` method